acl.js

```diff
@@ -284,11 +286,46 @@ function allowOrigin(mode, userId, res, aclGraph, subject) {
     }
 }
 
+function fetchDocument(graph, uri, req) {
+    var options = req.app.locals.ldp;
+    var uriBase = file.uriBase(req);
+
+    var body = "";
+    if (S(uri).startsWith(uriBase)) {
+        try {
+            var oldPath = req.url;
+            var newPath = S(uri).chompLeft(uriBase).s;
+            var documentPath = file.uriToFilename(newPath, options.root);
```

Comment:
It's not. New path is the path to the document being fetched (e.g. the
group file) while req.path is the path to the current request (e.g
ABC.ttl). The code needs to change the URL attribute of req because allow
takes a req object as its argument.
On Jul 17, 2015 8:54 AM, "Nicola Greco" <notifications@github.com> wrote:

> In acl.js
> <https://github.com/linkeddata/ldnode/pull/87#discussion_r34886032>:
>
> > @@ -284,11 +286,46 @@ function allowOrigin(mode, userId, res, aclGraph, subject) {
> >      }
> >  }
> >
> > +function fetchDocument(graph, uri, req) {
> > +    var options = req.app.locals.ldp;
> > +    var uriBase = file.uriBase(req);
> > +
> > +    var body = "";
> > +    if (S(uri).startsWith(uriBase)) {
> > +        try {
> > +            var oldPath = req.url;
> > +            var newPath = S(uri).chompLeft(uriBase).s;
> > +            var documentPath = file.uriToFilename(newPath, options.root);
>
> newPath is the equivalent of req.path, can we use that instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/linkeddata/ldnode/pull/87/files#r34886032>.
>